proposed/http-middleware/middleware-meta.md

```diff
@@ -1,49 +1,56 @@
-PSR-15 Meta Document
-====================
+HTTP Server Middleware
+======================
 
 1. Summary
 ----------
 
 The purpose of this PSR is to provide an interface that defines the formal
-method signature for HTTP Middleware that is compatible with HTTP Messages,
-as defined in PSR-7.
+method signature for HTTP Server Middleware that is compatible with HTTP
+Messages, as defined in [PSR-7][psr7].
+
+[psr7]: http://www.php-fig.org/psr/psr-7/
 
 2. Why Bother?
 --------------
 
-The general concept of reusable middleware was popularized by [StackPHP][stackphp].
-Since the release of the HTTP Messages standard, a number of frameworks have
-adopted middleware that uses HTTP Message interfaces.
+The HTTP Messages specification does not contain any reference to HTTP Middleware.
+
+The design pattern used by middleware has existed for many years as the
+[pipeline pattern][pipeline], or more specifically, "linear pipeline processing".
```

Comment:
> The Chain of Responsibility Pattern is well known for many many years, which uses only 1 interface (not 2!) and fit much much better into the OOP world (which supports my argument at i.).

This is a _similar_ pattern, and used in Rack, Stack, and several other middleware frameworks across different languages. It has a _qualitative difference_, however: it depends on either the interface defining a property (impossible in PHP), or a _convention_ of defining a property (impossible to dictate, and introduces variance between implementations); in each case, the _property_ is then accessed in order to delegate handling of the request. Additionally, it means that, from a practical standpoint, the top-level handler needs to manage a stack/queue/delegate/what-have-you internally.

The approach taken in this proposal is more along a _functional_ paradigm, requiring passage of a delegate as an _argument_. This allows a stronger, more concrete interface with no ambiguity and no reliance on _conventions_, which may differ from one implementation to the next. It is, in fact, why modern middleware stacks (Expressive, Slim 3, etc.) pass `$next` as an argument, versus relying on composition of the next middleware or middleware stack as a property. (This is also why usage of interfaces versus the looser `callable` is preferred, as it does not rely on a _convention_.)

Software design patterns are meant as a way of communicating how a system works, but can, and should, be customized or mixed with others to achieve a specific goal. That's exactly what is happening with this proposal.